evm/src/fixed_recursive_verifier.rs

```diff
@@ -1034,6 +1035,14 @@ where
                 .set_proof_with_pis_target(&self.block.parent_block_proof, parent_block_proof);
         } else {
             // Initialize genesis_state_trie, state_root_after, block hashes and the block number for correct connection between blocks.
+
+            if public_values.trie_roots_before.state_root
+                != public_values.extra_block_data.genesis_state_trie_root
+            {
+                return Err(anyhow::Error::msg(
+                    "Inconsistent pre-state for first block with genesis state.",
```

Comment:
Include the hashes in the error message? "Inconsistent pre-state for first block with genesis state. Expected {}, found {}" or so?

Reply:
I used `{:?}` format instead otherwise hashes would be pruned and only display first and last few bytes.